### PR TITLE
update TrackExtrapolatingAlg.cpp

### PR DESCRIPTION
**Pull request by Yang987915**
Evaluate the extrapolated points in each layer

### CODE REVIEW

File: Reconstruction/CRDEcalRec/src/Algorithm/TrackExtrapolatingAlg.cpp [hunk 1]
```diff
@@ -308,7 +307,7 @@ float TrackExtrapolatingAlg::GetRefAlpha0(const PandoraPlus::TrackState & trk_st
 // ...oooOO0OOooo......oooOO0OOooo......oooOO0OOooo...
 bool TrackExtrapolatingAlg::IsReturn(float rho, TVector2 & center){
   float farest = rho + center.Mod();
-  if (farest < settings.map_floatPars["outermost_distance"]) {return true;}
+  if (farest < settings.map_floatPars["outermost_distance"]/Cos(22.5/180.*Pi())+100) {return true;}
```
Comment (qwert2333) [inline Reconstruction/CRDEcalRec/src/Algorithm/TrackExtrapolatingAlg.cpp]: Add Nmodule=8 in settings, and replace the hardcoded 8 / 22.5 in all codes.

File: Reconstruction/CRDEcalRec/src/Algorithm/TrackExtrapolatingAlg.cpp [hunk 2]
```diff
@@ -140,7 +140,7 @@ StatusCode TrackExtrapolatingAlg::GetLayerPoints(const std::vector<TVector2> & n
     std::vector<TVector2> t_points;
     for(int il=0; il<settings.map_intPars["Nlayers"]; il++){
       TVector2 t_p;
-      float dist = settings.map_floatPars["layer_width"]*il + settings.map_floatPars["innermost_distance"];
+      float dist = settings.map_floatPars["layer_width"]*il + settings.map_floatPars["innermost_distance"] + 5;
```
Comment (qwert2333) [inline Reconstruction/CRDEcalRec/src/Algorithm/TrackExtrapolatingAlg.cpp]: Maybe should be layer_width/2 ?

File: Reconstruction/CRDEcalRec/src/Algorithm/TrackExtrapolatingAlg.cpp [hunk 3]
```diff
@@ -16,9 +16,9 @@ StatusCode TrackExtrapolatingAlg::ReadSettings(Settings& m_settings){
...
   // if(settings.map_floatPars.find("Par1")==settings.map_floatPars.end()) settings.map_floatPars["Par1"] = 0;
   if(settings.map_floatPars.find("innermost_distance")==settings.map_floatPars.end()) 
-    settings.map_floatPars["innermost_distance"] = 1865;
+    settings.map_floatPars["innermost_distance"] = 1860;
   if(settings.map_floatPars.find("outermost_distance")==settings.map_floatPars.end()) 
-    settings.map_floatPars["outermost_distance"] = 2400;
+    settings.map_floatPars["outermost_distance"] = 2140;
   if(settings.map_intPars.find("Nlayers")==settings.map_intPars.end()) 
     settings.map_intPars["Nlayers"] = 28;
   if(settings.map_floatPars.find("layer_width")==settings.map_floatPars.end())
```
Comment (qwert2333) [inline Reconstruction/CRDEcalRec/src/Algorithm/TrackExtrapolatingAlg.cpp:24]: For the following "normal_vec" value I missed last time: if they are necessary to be added in settings? (i.e. will you change them in any condition?)
If they are just normal initial value, you can write them in code by hardcoding.

**Review by qwert2333 — CHANGES_REQUESTED** [review]
Comments for the PR.
Global: review all your hardcoded numbers: write them in settings. (B-field, half-Z, etc.)

File: Reconstruction/CRDEcalRec/src/Algorithm/TrackExtrapolatingAlg.cpp (hunk 3 above)
Comment (Yang987915) [reply Reconstruction/CRDEcalRec/src/Algorithm/TrackExtrapolatingAlg.cpp:24]: These "normal_vec" are vectors that are perpendicular to each module.
They will be changed if the number of module changes or the whole ECAL rotate an angle.
If the ECAL with 8 modules do not change, "normal_vec" can be written as hardcode

Comment (qwert2333) [reply Reconstruction/CRDEcalRec/src/Algorithm/TrackExtrapolatingAlg.cpp:24]: Then you can use the module number as an option, and hardcode these values in the algorithm. 
```
if(settings.map_floatPar["Nmodule"]==8){
  xxxx;
}
```

**Pull request merged**